src/bindings/g_object/object.cr

```diff
@@ -20,7 +20,7 @@ module GObject
 
         def self.g_type : UInt64
           if LibGLib.g_once_init_enter(pointerof(@@_g_type)) != 0
-            g_type = {{ @type.superclass.id }}._register_derived_type({{ @type.id }},
+            g_type = {{ @type.superclass.id }}._register_derived_type("{{ @type.name.gsub(/::/, "-") }}",
```

Comment:
```suggestion
            g_type = {{ @type.superclass.id }}._register_derived_type("{{ @type.name.gsub("::", "-") }}",
```
gsub with Strings should be faster

Reply:
[That is not possible in macro land](https://crystal-lang.org/api/1.4.1/Crystal/Macros/MacroId.html#gsub%28regex%3ARegexLiteral%2Creplacement%3AStringLiteral%29%3AMacroId-instance-method)